build.assets/Dockerfile-babybox

```diff
@@ -0,0 +1,44 @@
+FROM golang:1.20
+
+# Image layers go from less likely to most likely to change.
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+    npm \
+    unzip \
+    && \
+  rm -rf /var/lib/apt/lists/*
+
+# protoc-gen-gogofast
+# Keep in sync with api/proto/buf.yaml (and buf.lock)
+ARG GOGO_PROTO_TAG=v1.3.2
+RUN go install "github.com/gogo/protobuf/protoc-gen-gogofast@$GOGO_PROTO_TAG"
+
+# protoc-gen-js and protoc-gen-ts
+ARG NODE_GRPC_TOOLS_VERSION=1.12.4
+ARG NODE_PROTOC_TS_VERSION=5.0.1
+RUN npm install --global "grpc-tools@$NODE_GRPC_TOOLS_VERSION" "grpc_tools_node_protoc_ts@$NODE_PROTOC_TS_VERSION"
+
+# protoc
+ARG PROTOC_VER=3.20.2
+RUN VERSION="$PROTOC_VER" && \
+  PB_REL='https://github.com/protocolbuffers/protobuf/releases' && \
+  cd /tmp && \
+  curl -fsSLO "$PB_REL/download/v$VERSION/protoc-$VERSION-linux-$(uname -m).zip"  && \
+  unzip protoc*.zip -d /usr/local && \
+  rm -f protoc*.zip
+
+# buf
+ARG BUF_VERSION=1.19.0
+RUN BIN='/usr/local/bin' && \
+  VERSION="$BUF_VERSION" && \
+  curl -fsSL "https://github.com/bufbuild/buf/releases/download/v$VERSION/buf-$(uname -s)-$(uname -m)" -o "${BIN}/buf" && \
+  chmod +x "$BIN/buf"
+
+# Pre-install go-runned binaries.
+# This is meant to be the only step that changes depending on the Teleport
+# branch.
+COPY go.mod go.sum /teleport-module/
+RUN cd /teleport-module; \
+  go install github.com/bufbuild/connect-go/cmd/protoc-gen-connect-go && \
+  go install google.golang.org/grpc/cmd/protoc-gen-go-grpc && \
+  go install google.golang.org/protobuf/cmd/protoc-gen-go
```

Comment:
would it make sense to switch these lines (can the `go install`s be cached)?

Reply:
Won't make much of a difference, as it's all in the same layer.

This step is the one bound to change the most, with go.mod updates and different branches versions and so on. It's here mostly so if you are doing codegen in the same branch in small intervals, then you won't have to build those binaries over and over. The buildbox solves that by binding GOMODCACHE to the host's /tmp, but this seems more idiomatic to me (and less prone to weird permissions errors).